snuba/cli/optimize.py

```diff
@@ -53,8 +48,23 @@ def optimize(
     table = enforce_table_writer(dataset).get_schema().get_local_table_name()
 
     today = datetime.utcnow().replace(hour=0, minute=0, second=0, microsecond=0)
-    clickhouse = ClickhousePool(
-        clickhouse_host, clickhouse_port, send_receive_timeout=timeout
-    )
-    num_dropped = run_optimize(clickhouse, database, table, before=today)
-    logger.info("Optimized %s partitions on %s" % (num_dropped, clickhouse_host))
+
+    if bool(clickhouse_host) ^ bool(clickhouse_port):
+        raise click.ClickException("Provide both Clickhouse host and port or neither")
+
+    # If Clickhouse host and port values are provided, run optimize on that host,
+    # otherwise run optimize on each registered cluster.
+    if clickhouse_host and clickhouse_port:
+        clickhouse_connections = [
+            ClickhousePool(
+                clickhouse_host, clickhouse_port, send_receive_timeout=timeout
+            )
+        ]
+    else:
+        clickhouse_connections = [
+            cluster.get_connection(send_receive_timeout=timeout) for cluster in CLUSTERS
+        ]
+
+    for connection in clickhouse_connections:
+        num_dropped = run_optimize(connection, database, table, before=today)
```

Comment:
Same concern I had on cleanup. we are running optimize on all cluster when the user provided specifically one dataset. What if one of the configured cluster is down. This would prevent us from doing optimize on a separate cluster that is unrelated.